### ISSUE TITLE
Evaluation metric for autolaparo

### ISSUE THREAD

**Issue by William-HYWu** [isyangshu/Surgformer#1]
Hi,
Thank you for your excellent work! I have two questions regarding the evaluation metric for AutoLaparo:

1. It seems like you only evaluated the test dataset (video 15-21), without using a validation dataset (11-14). Is that correct?
2. The paper mentions that you used unrelaxed evaluation for AutoLaparo. However, in `Main_autolaparo.m`, it calls the function in `Evaluate.m`, which applies relaxed boundary conditions. I'm wondering if I miss something in my implementation?

Thanks for your time and clarification!

**Comment by isyangshu** [comment]
For AutoLaparo, we use validation set for evaluation in the training
process. Since the training code and Evaluate.m used for both Cholec80 and
AutoLaparo, you need to change something, including using Val set on
training and remove related part.

Haoyang_Wu ***@***.***>于2024年10月11日 周五下午1:22写道：

**Comment by William-HYWu** [comment]
Hi,
Thank you for your response, and I apologize for not being clear in my previous message.

I wanted to ask specifically about the final evaluation process, after running `convert_cholec80/autolaparo.py` and the MATLAB code. From what I understand, the final results (the ones you included in the table in the paper) were obtained using _Main_autolaparo.m_ and _Main.m_.
However, in both _Main.m_ (line 56) and _Main_autolaparo.m_ (line 58), there’s a line of code that calls the Evaluate function:
`[jaccard, prec, rec, acc, f1] = Evaluate(gtLabelID, predLabelID, fps);` that calls the function Upon reviewing _Evaluate.m_, I noticed that it relaxes boundary conditions, especially for phases 4-7, as shown here:
``` 
       % relaxed boundary
       % revised for cholec80 dataset !!!!!!!!!!!
        if(iPhase == 4 || iPhase == 5) % Gallbladder dissection and packaging might jump between two phases
            curDiff(curDiff(1:t)==-1) = 0; % late transition
            curDiff(curDiff(end-t+1:end)==1 | curDiff(end-t+1:end)==2) = 0; % early transition % 5 can be predicted as 6/7 at the end > 5 followed by 6/7
        elseif(iPhase == 6 || iPhase == 7) % Gallbladder dissection might jump between two phases
            curDiff(curDiff(1:t)==-1 | curDiff(1:t)==-2) = 0; % late transition
            curDiff(curDiff(end-t+1:end)==1 | curDiff(end-t+1:end)==2) = 0; % early transition
        else
            % general situation
            curDiff(curDiff(1:t)==-1) = 0; % late transition
            curDiff(curDiff(end-t+1:end)==1) = 0; % early transition
        end
```
This seems to apply a relaxed evaluation for transitions between certain phases, which differs from the unrelaxed evaluation described in the paper.

Could you kindly clarify if this is intended or if I may have misunderstood the implementation?

Thank you again for your time and help—I greatly appreciate it!

**Comment by isyangshu** [comment]
For Cholec80, you need to manually turn the relaxation evaluation on and
off. For AutoLaparo, you need to remove the relaxed evaluation manually.
Currently, evaluate.m is not available for all datasets. We also remove
relaxed evaluation manually. We only use unrelated evaluation for
AUtoLaparo.

Haoyang_Wu ***@***.***>于2024年10月11日 周五下午5:21写道：

**Comment by William-HYWu** [comment]
Thanks, I understand it know
